python/lsst/daf/butler/core/mappingFactory.py

```diff
@@ -189,9 +190,14 @@ def getFromRegistryWithMatch(self, targetClasses: Iterable[Any], *args: Any,
         key, cls, registry_kwargs = self.getClassFromRegistryWithMatch(targetClasses)
 
         # Supplied keyword args must overwrite registry defaults
-        registry_kwargs.update(kwargs)
-
-        return key, cls(*args, **registry_kwargs)
+        # We want this overwriting to happen recursively since we expect
+        # some of these keyword arguments to be dicts.
+        # Simplest to use Config for this
+        config_kwargs = Config(registry_kwargs)
+        config_kwargs.update(kwargs)
+        merged_kwargs = dict(config_kwargs)
```

Comment:
This can't actually convert the inner mappings back to `dict` as well.  Does that just not matter?

Reply:
I don't think it matters because everything lower down just gets passed directly anyhow and `Config` behaves like a dict. It might be that that final line isn't really needed either.